packages/lit-html/src/directives/ref.ts

```diff
@@ -74,19 +74,19 @@ class RefDirective extends AsyncDirective {
       // which case it would otherwise be called with the new value before the
       // previous one unsets it) and down in the tree (where it would be unset
       // before being set). Note that element lookup is keyed by
-      // both the callback and the context, since we allow passing unbound
+      // both the context and the callback, since we allow passing unbound
       // functions that are called on options.host, and we want to treat
       // these as unique "instances" of a function.
-      let lastElementForContext = lastElementForCallback.get(this._ref);
+      const context = this._context ?? globalThis;
+      let lastElementForContext = lastElementForCallback.get(context);
       if (lastElementForContext === undefined) {
         lastElementForContext = new WeakMap();
-        lastElementForCallback.set(this._ref, lastElementForContext);
+        lastElementForCallback.set(context, lastElementForContext);
```

Comment:
(no action required - comment because I got a bit confused with the naming here but was able to figure it out.)

I think I found it surprising that `lastElementForCallback` has a context as key and in my mind returns a `lastElementForCallback` weakmap.

Could having the global map be called `lastElementForContextAndCallback`, then here it would return `lastElementForCallback`.

Then later (L:89) `lastElementForCallback.set(this._ref, element);` can be used.

Reply:
Yeah, making the names make sense with the double-keying is hard, but I think this is an improvement! Done.